commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -0,0 +1,207 @@
+		return err
+	}
+	return nil
+}
+
+// TODO: user specified tests for operators
+
+// checkStatusUpdate looks at all fields in the spec section of a custom resource and attempts to modify them and
+// see if the status changes as a result. This is a bit prone to breakage as this is a black box test and we don't
+// know much about how the operators we are testing actually work and may pass an invalid value. In the future, we
+// should use user-specified tests
+func checkStatusUpdate(runtimeClient client.Client, obj unstructured.Unstructured) error {
+	test := scorecardTest{testType: basicOperator, name: "Operator actions are reflected in status", maximumPoints: 1}
+	err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+	if err != nil {
+		return fmt.Errorf("error getting custom resource: %v", err)
+	}
+	if obj.Object["status"] == nil || obj.Object["spec"] == nil {
+		scTests = append(scTests, test)
+		return nil
+	}
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	specMap := obj.Object["spec"].(map[string]interface{})
+	err = modifySpecAndCheck(specMap, obj)
+	if err != nil {
+		test.earnedPoints = 0
+		scTests = append(scTests, test)
+		return nil
+	}
+	test.earnedPoints = 1
+	scTests = append(scTests, test)
+	return nil
+}
+
+// modifySpecAndCheck is a helper function for checkStatusUpdate
+func modifySpecAndCheck(specMap map[string]interface{}, obj unstructured.Unstructured) error {
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	var err error
+	for k, v := range specMap {
+		mapType := false
+		switch t := v.(type) {
+		case int64:
+			specMap[k] = specMap[k].(int64) + 1
+		case float64:
+			specMap[k] = specMap[k].(float64) + 1
+		case string:
+			// TODO: try and find out how to make this better
+			// Since strings may be very operator specific, this test may not work correctly in many cases
+			specMap[k] = fmt.Sprintf("operator sdk test value %f", rand.Float64())
+		case bool:
+			specMap[k] = !specMap[k].(bool)
+		case map[string]interface{}:
+			mapType = true
+			err = modifySpecAndCheck(specMap[k].(map[string]interface{}), obj)
+		case []map[string]interface{}:
+			mapType = true
+			for _, item := range specMap[k].([]map[string]interface{}) {
+				err = modifySpecAndCheck(item, obj)
+				if err != nil {
+					break
+				}
+			}
+		case []interface{}: // TODO: Decide how this should be handled
+		case interface{}: // same as above
+		default:
+			fmt.Printf("Unknown type for key (%s) in status: (%v)\n", k, reflect.TypeOf(t))
+		}
+		if !mapType {
+			runtimeClient.Update(context.TODO(), &obj)
+			err = wait.Poll(time.Second*1, time.Second*15, func() (done bool, err error) {
+				runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
```

Comment:
Check return error

Reply:
Also any reason why we can't use the object name/namespace e.g `obj.GetName()` rather than using the global var?

Reply:
The apiversion, kind and name is only parsed and noted for the CR manifest. I'm wondering if it isn't better to move that outside of `createFromYaml()` into a separate function and use that in `ScorecardTests()`.

We're pretty much using the name,apiversion and kind global vars as return arguments right now.
https://github.com/operator-framework/operator-sdk/blob/8c6a10411d8fa0e7c72e30d9a864c64af1f31408/commands/operator-sdk/cmd/scorecard/scorecard.go#L173-L175

That should simplify `createFromYaml` a bit as well. WDYT?

Reply:
That seems reasonable